addons/point_of_sale/models/pos_config.py

```diff
@@ -351,19 +351,14 @@ def create(self, values):
 
     @api.multi
     def write(self, vals):
-        pos_config_need_display = self.env['pos.config']
-        for pos_config in self:
-            if (pos_config.is_posbox or vals.get('is_posbox')) and (pos_config.iface_customer_facing_display or vals.get('iface_customer_facing_display')):
-                facing_display = (pos_config.customer_facing_display_html or vals.get('customer_facing_display_html') or '').strip()
-                if not facing_display:
-                    pos_config_need_display |= pos_config
+        result = super(PosConfig, self).write(vals)
 
-        if pos_config_need_display:
-            vals_display = vals.copy()
-            vals_display['customer_facing_display_html'] = self._compute_default_customer_html()
-            super(PosConfig, pos_config_need_display).write(vals_display)
+        default_customer_html = False
```

Comment:
Maybe it is shorter like this and avoid to loop on filtered records that is a loop:
config_display = self.filtered(lambda c: c.is_posbox and c.iface_customer_facing_display and not (c.customer_facing_display_html or '').strip())
if config_display:
    super(PosConfig, config_display).write({'customer_facing_display_html': self._compute_default_customer_html()})